core/commands/object/object.go

```diff
@@ -566,9 +578,14 @@ func getOutput(dagnode *dag.Node) (*Object, error) {
 }
 
 // converts the Node object into a real dag.Node
-func deserializeNode(node *Node) (*dag.Node, error) {
+func deserializeNode(node *Node, dataFieldEncoding string) (*dag.Node, error) {
 	dagnode := new(dag.Node)
-	dagnode.Data = []byte(node.Data)
+	if dataFieldEncoding == "text" {
```

Comment:
should do a switch here, so we don't accidentaly get base64 encoding by passing `"cats"`